modules/core/src/main/scala/org/tessellation/domain/trust/storage/OrdinalTrustMap.scala

```diff
@@ -0,0 +1,35 @@
+package org.tessellation.domain.trust.storage
+
+import cats.Order
+import cats.syntax.contravariant._
+
+import org.tessellation.cli.method.snapshotInterval
+import org.tessellation.schema.SnapshotOrdinal
+
+import derevo.circe.magnolia.encoder
+import derevo.derive
+
+@derive(encoder)
+case class OrdinalTrustMap(
+  ordinal: SnapshotOrdinal,
+  peerLabels: PublicTrustMap,
+  trust: TrustMap
+)
+
+object OrdinalTrustMap {
+
+  implicit val order: Order[OrdinalTrustMap] = Order[SnapshotOrdinal].contramap(_.ordinal)
+
+  implicit def catsOrder(implicit ord: Ordering[OrdinalTrustMap]): Order[OrdinalTrustMap] = ord.compare(_, _)
+
+  def empty: OrdinalTrustMap = OrdinalTrustMap(SnapshotOrdinal.MinValue, PublicTrustMap.empty, TrustMap.empty)
+
+  def initial: OrdinalTrustMap = OrdinalTrustMap(SnapshotOrdinal(snapshotInterval), PublicTrustMap.empty, TrustMap.empty)
+
+  def advanceInterval(trust: OrdinalTrustMap): OrdinalTrustMap = OrdinalTrustMap(
+    trust.ordinal.plus(snapshotInterval),
+    PublicTrustMap.empty,
+    TrustMap.empty
+  )
```

Comment:
shouldn't `advanceInterval` also set the public trust map and trust map?
If the intended algorithm is to firstly update the interval, then update the rest, imo it should be:
```
ordinal: Ordinal,
trust: Option[(PublicTrustMap, TrustMap)]
```
or an equivalent case class. anyway, to distinguish "no data yet" from "data provided - they are empty"

Reply:
Every `snapshotInterval` + `delayToUpdateCurrent` ordinals, the `next` `OrdinalTrustMap` is made the `current` `OrdinalTrustMap`. At that point, `next` is emptied so `next.trust` and `next.peerLabels` are made emptied. `next.ordinal` is advanced to the next ordinal that `next` will be updated (`snapshotInterval` is added to `next.ordinal`).